crates/kas-widgets/src/menu/menubar.rs

```diff
@@ -107,46 +107,45 @@ widget! {
                     coord,
                     ..
                 } => {
-                    mgr.set_grab_depress(source, cur_id.clone());
-                    if let Some(id) = cur_id {
-                        if self.bar.is_strict_ancestor_of(&id) {
-                            // We instantly open a sub-menu on motion over the bar,
-                            // but delay when over a sub-menu (most intuitive?)
-                            if self.rect().contains(coord) && !self.bar.eq_id(&id) {
-                                self.set_menu_path(mgr, Some(&id), false);
-                            } else if id != self.delayed_open {
-                                mgr.set_nav_focus(id.clone(), false);
-                                let delay = mgr.config().menu_delay();
-                                mgr.update_on_timer(delay, self.id(), id.as_u64());
-                                self.delayed_open = Some(id);
-                            }
-                        }
+                    if !mgr.set_grab_depress(source, cur_id.clone()) {
+                        // If redraw hasn't happened, force it to fix
+                        // highligting of previously selected menu-item
+                        mgr.send_action(TkAction::REDRAW);
```

Comment:
This doesn't sound right — if `set_grab_depress` returns `false`, the depressed widget hasn't changed. Menu entries are highlighted if either the `depress` or the `nav_focus` state is true. A change to `nav_focus` should also trigger a redraw.

I can reproduce the issue even with this extra redraw command, though in this case mouse movement must be stopped immediately after the mouse touches the new menu title, or a redraw happens.

My guess: the issue is that `SubMenu::draw` explicitly sets `DEPRESS` when `self.popup_id.is_some()` — sending of `Event::PopupRemoved` is delayed slightly.

Reply:
Better fix (possibly handling `PopupRemoved` is redundant here; probably still better to do so):
```rust
// submenu.rs

        fn close_menu(&mut self, mgr: &mut EventMgr, restore_focus: bool) {
            if let Some(id) = self.popup_id {
                mgr.close_window(id, restore_focus);
                self.popup_id = None;
            }
        }

        fn handle(&mut self, mgr: &mut EventMgr, event: Event) -> Response<M> {
            match event {
                // ...
                Event::PopupRemoved(id) => {
                    if self.popup_id.as_ref() == Some(&id) {
                        self.popup_id = None;
                    }
                    Response::Used
                }
```
Or we could simply request a redraw in `PopupRemoved` (though this may redraw twice, flashing the double-highlight).

Reply:
Causing a redraw here should not be needed now.